integration/app.py

```
@@ -80,88 +81,49 @@ def create_check():
-            logger.info("BGC adapter timeout", extra=get_logger_data(check_data))
-            return jsonify(
-                Response(
-                    status=FAILED,
-                    error_details=[
-                        ErrorDetail(
-                            code="408", message=json.loads({"error": "Timeout"})
-                        )
-                    ],
-                )
-            )
-        except (BadRequestAPIException, NotFoundAPIException) as e:
-            logger.info(
-                "BGC adapter request exception",
-                extra=get_logger_data(check_data, e.message),
-            )
-            return jsonify(
-                Response(
-                    status=FAILED,
-                    error_details=[
-                        ErrorDetail(code="400", message=json.loads(e.message))
-                    ],
-                )
-            )
+            return response_data
         except GenericAPIException as e:
-            logger.info(
-                "BGC adapter generic exception",
-                extra=get_logger_data(check_data, e.message),
-            )
-            return jsonify(
-                Response(
-                    status=FAILED,
-                    error_details=[
-                        ErrorDetail(code="500", message=json.loads(e.message))
-                    ],
-                )
-            )
-        return response_data
+            logger.info(f"BGC request error {e.error_message}", extra=get_logger_data(check_data, e.error_message))
+            return get_error_response(e, 400)
 
     @app.route("/get_check", methods=["POST"])
     def get_check():
-        signature = request.headers.get("Authorization")
-        password = str(base64.b64decode(signature), "utf-8")
-
-        if password != getenv("REQUEST_PASSWORD"):
-            return json.dumps({"success": False}), 403
+        if not validate_request(request.headers.get("Authorization")):
```

this service `validate_request` should raise the exception and you should catch it here and send to get_error_response the caught exception. maybe the exceptions for the satellite should have the status code to be returned. so you don't need to pass it, and you could be able to raise different exceptions that returns  different status codes